app/modules/session/session.server.ts

```diff
@@ -69,3 +69,11 @@ export async function logout(request: Request) {
     },
   });
 }
+
+export async function requireUserSession(request: Request) {
+  const session = await getUserSession(request);
+  if (!session) {
+    return redirect('/login');
```

Comment:
```suggestion
    throw redirect('/login');
```

Reply:
Context:
- https://sergiodxa.com/articles/throwing-vs-returning-responses-in-remix
- https://remix.run/docs/en/main/guides/errors